shader/path_tiling.wgsl

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT OR Unlicense
+
+// Write path segments
+
+#import bump
+#import config
+#import segment
+#import tile
+
+@group(0) @binding(0)
+var<storage> bump: BumpAllocators;
+
+@group(0) @binding(1)
+var<storage> seg_counts: array<SegmentCount>;
+
+@group(0) @binding(2)
+var<storage> lines: array<LineSoup>;
+
+@group(0) @binding(3)
+var<storage> paths: array<Path>;
+
+@group(0) @binding(4)
+var<storage> tiles: array<Tile>;
+
+@group(0) @binding(5)
+var<storage, read_write> segments: array<Segment>;
+
+fn span(a: f32, b: f32) -> u32 {
+    return u32(max(ceil(max(a, b)) - floor(min(a, b)), 1.0));
+}
+
+// One invocation for each tile that is to be written.
+// Total number of invocations = bump.seg_counts
+@compute @workgroup_size(256)
+fn main(
+    @builtin(global_invocation_id) global_id: vec3<u32>,
+) {
+    let n_segments = atomicLoad(&bump.seg_counts);
+    if global_id.x < n_segments {
+        let seg_count = seg_counts[global_id.x];
+        let line = lines[seg_count.line_ix];
+        let counts = seg_count.counts;
+        let seg_within_slice = counts >> 16u;
+        let seg_within_line = counts & 0xffffu;
+
+        // coarse rasterization logic
+        let is_down = line.p1.y >= line.p0.y;
+        var xy0 = select(line.p1, line.p0, is_down);
+        var xy1 = select(line.p0, line.p1, is_down);
+        let s0 = xy0 * TILE_SCALE;
+        let s1 = xy1 * TILE_SCALE;
+        let count = span(s0.x, s1.x) + span(s0.y, s1.y) - 1u;
+        let dx = abs(s1.x - s0.x);
+        let dy = s1.y - s0.y;
+        let idxdy = 1.0 / (dx + dy);
```

Comment:
Maybe document that `dx + dy` cannot be zero and the division is valid because a zero-length line should have been discarded by `path_count`?